### PR TITLE
declare, local, typeset: add pages

### PR DESCRIPTION
**Pull request by EmilySeville7cfg**
- [x] The page (if new), does not already exist in the repository.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

**Version of the command being documented (if known):**

### CODE REVIEW

File: pages/common/declare.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+- Declare an associative array variable:
+
+`declare -A {{variable}}=({{[key_a]=item_a [key_b]=item_b [key_c]=item_c}})`
+
+- Declare a readonly string variable:
+
+`declare -r {{variable}}="{{value}}"`
+
+- Declare a global variable even within function:
```
Comment (sbrl) [inline pages/common/declare.md:26]: Perhaps this will read better?
```suggestion
- Declare a global variable within a function:
```

**Review by sbrl — COMMENTED** [review]
Thanks for the new pages! I've left a comment below for you to review.

File: pages/common/declare.md (hunk 1 above)
Comment (EmilySeville7cfg) [reply pages/common/declare.md]: I've fixed everything at once.

**Review by sbrl — APPROVED** [review]
Thanks, @EmilySeville7cfg!

File: pages/common/declare.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+- Declare an associative array variable with the specified value:
+
+`declare -A {{variable}}=({{[key_a]=item_a [key_b]=item_b [key_c]=item_c}})`
+
+- Declare a readonly string variable with the specified value:
+
+`declare -r {{variable}}="{{value}}"`
+
+- Declare a global variable even within a function with the specified value:
```
Comment (navarroaxel) [inline pages/common/declare.md:26]:
```suggestion
- Declare a global variable within a function with the specified value:
```

File: pages/common/typeset.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+- Declare an associative array variable with the specified value:
+
+`typeset -A {{variable}}=({{[key_a]=item_a [key_b]=item_b [key_c]=item_c}})`
+
+- Declare a readonly variable with the specified value:
+
+`typeset -r {{variable}}="{{value}}"`
+
+- Declare a global variable even within a function with the specified value:
```
Comment (navarroaxel) [inline pages/common/typeset.md:26]:
```suggestion
- Declare a global variable within a function with the specified value:
```

**Review by navarroaxel — APPROVED** [review]

**Comment by EmilySeville7cfg** [comment]
Why there is no `mass-changes` tag?

**Pull request merged**